### ISSUE TITLE
query_pan_genome ends with out an error and an empty output file

### ISSUE THREAD

**Issue by ga23981** [sanger-pathogens/Roary#455]
/usr/local/apps/eb/Roary/3.12.0/bin/query_pan_genome -a gene_multifasta -n fhuE,mog,adaA,group_3839,mgtC,group_14860,trg_11,group_1011,group_1401,atzC,yacG,ushA,group_2341,group_3976,mdoC,ycgZ_1,group_4062,galR,group_1745,lpxH,amiA,group_1714,cynR_2,group_4197,group_1141,fecA,pgrR_3,group_3843,group_842,phoA,uxuA,ispB,dhaM,group_2145,group_2845,fruA_4,srrA,phnM,group_3896,yxeP_1,fhuA_1,atoC_1,inhA,group_2209,ygiC,group_3919,group_2502,motB_2,group_1463,ycfS,gatA_1,group_3819,damX,group_375,group_1702,group_3883,carB,proY,group_1481,lplT,ampC_2,oppF_2,group_2731,yihQ,dlgD,ispH,group_1509,group_2856,entS,group_1057,group_2620,rbsR_1,group_2397,zntB_1,yicJ_3,nagC_6,group_14857,yihS,betT,group_1584,araD,pdeD,atzE,galP,arsC,phoC,por_2,argE_1,btuR,tap_1,ygcS,fimD,group_1405,mrcB,pheC,group_2261,ampH,patB_1,group_3794,pgl_1,pcaK_2,iolS_2,group_3822,lptE,yihP_2,soxS_2,tnaB,znuC,hemH,ribH2,nfrA2,putA,group_2650,group_2367,rebM,soxC,benM_2,pucD,ddpF_2,aspS,group_4068,ydjH,group_14117,pcaF,nuoG,ybjG_2,gldA,rluA_1,hmuU_2,nadK,ttdT,group_784,cyaA,group_2316,group_3825,tsr_4,group_1007,lptC,ghrA_1,trg_8,group_13910,ygiN,moeB_2,xylB_2,pgl_3,group_3972,group_706,group_3835,group_1684,nasA,lpxT,hisI,dhaA,bdlA_1,group_860,yhaJ_1,yisK,ynaI_1,group_598,sbmA,group_1434,group_1505,tus,group_4176,betI_2,lsrR,catD,srlR_2,pstS_1,luxA,group_2490,ascG_2,ydeP_1,amaB,yciV,yihR,dapA_1,budB,dmoA,serS,group_990,group_1537,fhuC_2,msuD,kefF_2,yveA,group_13585,group_3897,group_3795,group_2415,astD,mexB_1,fbaA_5,allS_2,hmuU_1,moaD,virS,kdgK_2,iolS_1,blh,group_2135,group_4180,yjhB_2,group_4178,ddc_2,group_1150,valS,pyrC,mutT,yfgD,xylA_1,yecD,group_2211,gspE,ssuC_2,manR_1,repB,ptrA_1,bmr3,group_2726,group_3834,cirA,group_3950,lgoT_1,iscS,otsA_2,glgB,group_14855,group_3848,dsbD_2,rpoD,tauD,group_283,arsR1,group_1615,arsB,proQ,rpnA_4,group_4072,group_10424,cimH,group_4194,group_3845,cph2,lptG,group_1036,pimB,group_14389,glpK_1,group_4439,group_4038,intA_3,fumD,ssuA_1,cfa,yvbK,group_1909,group_2152,hosA,nimR_2,lolD_2,gatY,group_3840,ecdB,yecN,rraB,dacC,glnH_1,ychF,glnP,ribB,edcC,wecC,group_3918,mngA_1,manP_1,group_2539,group_1594,yycE,cynT,ygiD,nemR_2,bsdD,cex,yqjG,group_4082,mngB,group_2400,group_12999,licR_1,group_3836,otnC,group_509,rutD_1,scrK,group_4084,icsA,group_829,psiF,mug,group_1717,group_1548,group_403,group_4043,group_4314,group_845,lamB_1,group_1134,group_1467,rl5,group_548,group_1635,ble,btr,group_1534,group_3095,auaH,malT,group_2744,atl,rhaA,rhaB,group_1788,cnbH,rppH_1,group_2176,group_2707,group_658,group_4374,yjiG_2,group_698,aaeB_2,group_3941,rhaD,group_3963,tsr_7,group_1626,decR_2,group_986,nicS,rhaM,coaE_2,rhaS_1,leuE,group_2215,group_2391,satP,rhaR,fruA_2,rhaT,group_2291,glpQ *.gff > soft_core_seq.fa

**Comment by ga23981** [comment]
The script runs for a while and generates an empty output file (soft_core_seq.fa) with out generating any error. Can anyone please suggest what wrong could I be doing here?

**Comment by tseemann** [comment]
Is it possible it ran successfully and that there are *no* genes matching your query?

**Comment by ga23981** [comment]
No, it  is not possible! The temp that it is generating contains the
protein sequences corresponding to those ids. But temp file exists as long
as the script runs and then it vanishes when the script ends. Also, when I
check the temp file when the program is still running i see only the
proteins in it but not the nucleotide seq.
I am working on a University cluster.
Thanks
Gaurav

**Comment by tseemann** [comment]
Run your command with ONE gene and see what happens. 
Then paste the full exact command you ran here if it doesn't work.

**Comment by ga23981** [comment]
Tried with one gene as following:
/usr/local/apps/eb/Roary/3.12.0/bin/query_pan_genome -a gene_multifasta -o sof_cor_gen_fa -n fhuE  *.gff
No output file is resulted.

**Comment by ga23981** [comment]
Sorry for my delayed response.

Tried with one gene as following:
/usr/local/apps/eb/Roary/3.12.0/bin/query_pan_genome -a gene_multifasta -o
sof_cor_gen_fa -n fhuE *.gff
No output file is resulted. No error message.

Gaurav

**Comment by ga23981** [comment]
If needed I can share my files.

**Comment by chodarq** [comment]
I have the same problem. The only option for query_pan_genome which not work is multifasta. It creates the temporal folders while running, but once finished, folders are erased and no multifasta sequence is generated.

**Comment by tseemann** [comment]
I note the front page says this:

![image](https://user-images.githubusercontent.com/453972/69849619-5747a780-12d1-11ea-8cce-ceca663921ba.png)

**Comment by chodarq** [comment]
I see. I arrive at this thread googling, not from the front page of Roary. 
Thanks!
